stacker/actions/destroy.py

```diff
@@ -0,0 +1,64 @@
+import logging
+
+from . import base
+from ..plan import Plan
+
+logger = logging.getLogger(__name__)
+
+
+class Action(base.BaseAction):
+
+    def _get_dependencies(self, stacks_dict):
+        dependencies = {}
+        for stack_name, stack in stacks_dict.iteritems():
+            provider_stack = self.provider.get_stack(stack.fqn)
+            required_stacks = []
+            if provider_stack:
+                required_stacks = self.provider.get_required_stacks(provider_stack)
+
+            if not required_stacks:
+                if stack_name not in dependencies:
```

Comment:
If the stack is already in dependencies, is ignoring it the right thing?  Might want some debug log statements in here.